exchanges/okcoin/okcoin_test.go

```diff
@@ -77,1251 +80,1670 @@ func TestStart(t *testing.T) {
-	feeBuilder.FiatCurrency = currency.USD
-	if _, err := o.GetFee(context.Background(), feeBuilder); err != nil {
+	_, err := o.GetOrderHistory(context.Background(), &getOrdersRequest)
+	if err != nil {
 		t.Error(err)
 	}
 }
 
-func TestFormatWithdrawPermissions(t *testing.T) {
+func TestGetFundingHistory(t *testing.T) {
 	t.Parallel()
-	expectedResult := exchange.AutoWithdrawCryptoText + " & " + exchange.NoFiatWithdrawalsText
-	withdrawPermissions := o.FormatWithdrawPermissions()
-	if withdrawPermissions != expectedResult {
-		t.Errorf("Expected: %s, Received: %s", expectedResult, withdrawPermissions)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, o)
+	_, err := o.GetFundingHistory(context.Background())
+	if err != nil {
+		t.Error(err)
 	}
 }
 
-// Wrapper tests --------------------------------------------------------------------------------------------------
-
 func TestSubmitOrder(t *testing.T) {
 	t.Parallel()
-	sharedtestvalues.SkipTestIfCannotManipulateOrders(t, o, canManipulateRealOrders)
-
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, o, canManipulateRealOrders)
 	var orderSubmission = &order.Submit{
-		Exchange: o.Name,
-		Pair: currency.Pair{
-			Base:  currency.BTC,
-			Quote: currency.USD,
-		},
+		Pair:      spotTradablePair,
+		Exchange:  o.Name,
 		Side:      order.Buy,
 		Type:      order.Limit,
-		Price:     -1,
-		Amount:    1,
-		ClientID:  "meowOrder",
+		Price:     1,
+		Amount:    1000000000,
+		ClientID:  "yeneOrder",
 		AssetType: asset.Spot,
 	}
-	response, err := o.SubmitOrder(context.Background(), orderSubmission)
-	if sharedtestvalues.AreAPICredentialsSet(o) && (err != nil || response.Status != order.New) {
-		t.Errorf("Order failed to be placed: %v", err)
-	} else if !sharedtestvalues.AreAPICredentialsSet(o) && err == nil {
-		t.Error("Expecting an error when no keys are set")
+	_, err := o.SubmitOrder(context.Background(), orderSubmission)
+	if err != nil {
+		t.Error(err)
 	}
 }
 
-func TestCancelExchangeOrder(t *testing.T) {
+func TestModifyOrder(t *testing.T) {
 	t.Parallel()
-	sharedtestvalues.SkipTestIfCannotManipulateOrders(t, o, canManipulateRealOrders)
-
-	var orderCancellation = order.Cancel{
-		OrderID:       "1",
-		WalletAddress: core.BitcoinDonationAddress,
-		AccountID:     "1",
-		Pair:          spotCurrency,
-	}
-
-	err := o.CancelOrder(context.Background(), &orderCancellation)
-	if !sharedtestvalues.AreAPICredentialsSet(o) && err == nil {
-		t.Error("Expecting an error when no keys are set")
-	}
-	if sharedtestvalues.AreAPICredentialsSet(o) && err != nil {
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, o, canManipulateRealOrders)
+	_, err := o.ModifyOrder(context.Background(),
+		&order.Modify{
+			AssetType: asset.Spot,
+			Pair:      spotTradablePair,
+			OrderID:   "1234",
+			Price:     123456.44,
+			Amount:    123,
+		})
+	if err != nil {
 		t.Error(err)
 	}
 }
 
-func TestCancelAllExchangeOrders(t *testing.T) {
+func TestCancelOrder(t *testing.T) {
```

Comment:
This is because the web-socket endpoint sends a short message like "Operation failed" instead of "order not found" or such similar error messages. Same for the above. After I fix the **_CancelBatchOrder_** method, It is returning `okcoin websocket error code: 1 message: Operation failed.` without a specific error message.

Reply:
I brought these up because the other functions were much more specific about not being able to find the order rather than operation failed. I can reverify, but my conclusion was that anything with that error had something else going wrong

Reply:
I tried those endpoints with an `"Operation failed"` by disabling the WebSocket; the errors were like `order not found` or `insufficient Balance` and some similar messages.

If needed I can also add unit test which tests the REST endpoint by disabling the web-socket

Reply:
Thank you for investigating! That won't be necessary so long as the functions that the wrapper functions call have their own tests which looks to be the case 🤙

Reply:
Actually, there's sub error messages for websocket issues:
```
{"id":"189817431","op":"order","code":"1","msg":"","data":[{"tag":"","ordId":"","clOrdId":"yeneOrder","sCode":"51008","sMsg":"Order failed. Insufficient USD balance in account"}]}
```
Could you include the sMsg error in the response rather than just the `code` of `1`?

Reply:
Ohh, I don't know why I didn't see this.
Change applied. Thanks!